### ISSUE TITLE
[Question] Question regarding the implementation of PyTorchWrapper

### ISSUE THREAD

**Issue by thomasryck** [mlflow/mlflow#1702]
I have a question regarding the following line of code

```
predicted.index = data.index
```
which is in the init file of the pytorch module.

```
class _PyTorchWrapper(object):
    """
    Wrapper class that creates a predict function such that
    predict(data: pd.DataFrame) -> model's output as pd.DataFrame (pandas DataFrame)
    """
    def __init__(self, pytorch_model):
        self.pytorch_model = pytorch_model

    def predict(self, data, device='cpu'):
        import torch

        if not isinstance(data, pd.DataFrame):
            raise TypeError("Input data should be pandas.DataFrame")
        self.pytorch_model.to(device)
        self.pytorch_model.eval()
        with torch.no_grad():
            input_tensor = torch.from_numpy(data.values.astype(np.float32)).to(device)
            preds = self.pytorch_model(input_tensor)
            if not isinstance(preds, torch.Tensor):
                raise TypeError("Expected PyTorch model to output a single output tensor, "
                                "but got output of type '{}'".format(type(preds)))
            predicted = pd.DataFrame(preds.numpy())
            predicted.index = data.index
            return predicted

```

This line of code shows implies that the input data and the output data of the network have the same indexes. However in my opinion is not the main case and the index of the prediction dataframe should be the indexes from the dataframe creation. 

This line of code is not compatible with my use case. My input shape is (3, X) and output is (2, X) .

Can someone explain me why the indexes are pass from input data to prediction ?

**Comment by smurching** [comment]
@thomasryck I think the original intention was that each row of the predicted result would have the same label as the corresponding input row. To confirm, is your model intended to produce a different number of predictions than inputs? Or is it that the number of predictions and inputs the same, but the inputs & predictions are 2D tensors of differing shapes (input shape is (3, X) and output is (2, X))?

Thanks!

**Comment by thomasryck** [comment]
@smurching My number of predictions and inputs are the same but they have different shapes. My input correspond to 3 channel images. And my output is 2 different score for each pixel.

**Comment by thomasryck** [comment]
@smurching I think that the two affirmation could be true. What I mean is that my input is a 3 channel image and my output is 2 different score for each pixel. So I have, as an output, 2*X predictions. So the number of predictions is different from the number of inputs. But I could also have only one score for each pixel which would give the same number of predictions and inputs but 2 different shapes (input (3,X) and output (1, X))
